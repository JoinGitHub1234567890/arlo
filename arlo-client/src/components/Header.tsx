import React from 'react'
import styled from 'styled-components'
import { api } from './utilities'

const HeaderContainer = styled.div`
  padding: 20px;
  text-align: center;
`
const ButtonBar = styled.div`
  display: inline-block;
  float: right;
`

const Button = styled.button`
  margin: 0 auto;
  border-radius: 5px;
  background: rgb(211, 211, 211);
  width: 155px;
  height: 30px;
  color: #000000;
  font-size: 0.4em;
  font-weight: 500;
`

const HomeImg = styled.img`
  position: absolute;
  top: 35%;
  left: 50%;
  transform: translateX(-50%);
`

const Header = (props: any) => {
  const reset = async () => {
    await api(`/audit/reset`, { method: 'POST' })

    // ugly but works
    window.location.reload()
  }

  return (
    <HeaderContainer>
      {props.isHome ? (
        <HomeImg height="60px" src="/arlo.png" alt="Arlo, by VotingWorks" />
      ) : (
        <img height="60px" src="/arlo.png" alt="Arlo, by VotingWorks" />
      )}
      {!props.isHome && (
        <ButtonBar>
          <Button onClick={reset}>Clear & Restart</Button>
        </ButtonBar>
      )}
    </HeaderContainer>
  )
}

export default Header
