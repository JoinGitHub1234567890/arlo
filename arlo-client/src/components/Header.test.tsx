import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Header from './Header'

it('renders correctly at /home', () => {
  const { container, queryByText } = render(<Header match={null} isHome />)
  expect(container).toMatchSnapshot()
  expect(queryByText('Clear & Restart')).toBeFalsy()
  expect(container).toMatchSnapshot()
})

it('renders correctly at /election', () => {
  const { container, getByText } = render(<Header match={null} />)
  expect(container).toMatchSnapshot()
  fireEvent.click(getByText('Clear & Restart'))
  expect(container).toMatchSnapshot()
})
