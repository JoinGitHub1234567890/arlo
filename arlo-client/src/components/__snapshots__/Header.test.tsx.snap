// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa cdOnQX"
  >
    <img
      alt="Arlo, by VotingWorks"
      height="60px"
      src="/arlo.png"
    />
    <div
      class="sc-bwzfXH iHMWje"
      id="reset-button-wrapper"
    />
  </div>
</div>
`;
